test/parallel/test-http2-compat-expect-continue-check.js

```diff
@@ -0,0 +1,71 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const testResBody = 'other stuff!\n';
+const testResHeaderKey = 'abcd';
+const testResHeaderVal = '1';
+
+function handler(req, res) {
+  console.error('Server sent full response');
+
+  res.writeHead(200, {
+    'content-type': 'text/plain',
+    [testResHeaderKey]: [testResHeaderVal]
+  });
+  res.end(testResBody);
+}
+
+const server = http2.createServer(
+  common.mustNotCall('Full response sent before 100 Continue')
+);
+
+server.on('checkContinue', common.mustCall((req, res) => {
+  console.error('Server received Expect: 100-continue');
+
+  res.writeContinue();
+
+  setTimeout(common.mustCall(() => handler(req, res)), 100);
```

Comment:
why the `setTimeout` here? If it's needed to defer for the I/O cycle, can you add a comment about it? 

@jasnell do we have a better way to do this? How could we know it has finished writing it?

Reply:
Will add a comment. Not sure there's a better way to allow 'continue' to fire on the client before we continue writing.